### DISCUSSION TITLE
Webpack5 prevent fallback packages from being resolved when config is required in application

### DISCUSSION THREAD

**Discussion by gnoo** [webpack/webpack#15846] [Q&A] [answered]
In Webpack 5, polyfill of core node modules are removed, instead required packages are required to listed in resolve.fallback property. Below is resolve property of `webpack.config.client.js` file.

```
resolve: {
  fallback: {
    path: require.resolve("path-browserify"),
    crypto: require.resolve("crypto-browserify"),
    "babel-polyfill": require.resolve("@babel/polyfill"),
    buffer: require.resolve("buffer/"),
    stream: require.resolve("stream-browserify"),
  }
}
```

I use webpack-dev-middleware and webpack-hot-middleware along with express to achieve benefit of HMR for my ssr application.

```
import express from "express";
import webpack from "webpack";
...
const app = express();
...
const webpackDevConfig = require("../webpack/webpack.config.client");
const compiler = webpack(webpackDevConfig);
app.use(
  require("webpack-dev-middleware")(compiler, {
    publicPath: webpackDevConfig.output.publicPath,
  }),
);
app.use(require("webpack-hot-middleware")(compiler));
...
```

When the config file is imported, those modules in `fallback property` are resolved and returned as number. And webpack throws an error when the config object is passed to its constructor since `configuration.resolve.fallback` property should be non empty string, but numbers were given.

Below is the actual resolve.fallback property passed, and the error returned.

```
{
  path: 79936 (which suppose to be "/some path/node_module/path/index.js",
  crypto: 32640,
  'babel-polyfill': 71202,
  buffer: 30816,
  stream: 78787
}

ValidationError: Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.resolve.fallback should be one of these:
   [object { alias, name, onlyModule? }, ...] | object { <key>: [non-empty string, ...] | false | non-empty string }
   -> Redirect module requests. 
.....
```

Is there any possible workaround for this issue?

**Comment by vankop** [comment, marked as answer]
looks weird.. I think you bundle your webpack config.. `require.resolve` can return number only when bundling by webpack..

**Reply by gnoo** [reply]
yeah, but to use [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware#usage), i need to import webpack config as compiler object to `app.js` file, which makes webpack config object to be bundled with `app.js` file, then library imported within config file is bundled too.

**Reply by vankop** [reply]
not sure what you mean.. from your code:
```js
const webpackDevConfig = require("../webpack/webpack.config.client");
const compiler = webpack(webpackDevConfig);
```
this means webpack config was just required.. or this code should be bundled? if so as I said `require.resolve` will be resolved by webpack. I think you can try to use `require.resolve(/* webpackIgnore: true */"blabla")`

**Reply by gnoo** [reply]
The magic comment(/`* webpackIgnore: true */`) works. Thanks!
